### PR TITLE
TREAS-22 Added mobile and tablet support for ideas form.

### PR DESCRIPTION
**Pull request by Milszym**

### CODE REVIEW

**Review by aolszewska — COMMENTED** [review]
image
https://xd.adobe.com/view/0bb66d6e-5797-40bc-aa29-c6f71131f402-5ee5/screen/702412d6-0b7d-4669-86d3-73afc656abbd/specs/
I think the margins are too large on mobile now. 
Another thing I have noticed is that the create idea form on the designs looks like a popup window with an `x` button to close it. Do you think we want to strictly follow the designs? Or shall we leave it as is and (maybe) implement it later?

**Comment by Milszym** [comment]
You're right, I have adjusted margins for mobile as well. I have also noticed that size of the font was not correct. I have changed it to always match `pixels` not `em`s.

I have added cross icon that navigates back when clicked. I was trying to create svg button, but it is not centered. Have you got any experience in such svg buttons in Material-UI library? I couldn't center the `cross` icon :/ 

Also I am not sure if wrapping SvgIcon with IconButton is the right way to go with material ui, couldn't found out another way though :/  What do you think?

Screenshot 2020-12-04 at 12 56 05

**Comment by aolszewska** [comment]
@Milszym 

> I have added cross icon that navigates back when clicked. I was trying to create svg button, but it is not centered. Have you got any experience in such svg buttons in Material-UI library? I couldn't center the cross icon :/
> 
> Also I am not sure if wrapping SvgIcon with IconButton is the right way to go with material ui, I couldn't find out another way though :/ What do you think?
I was using a pure HTML `img` tag for the svg icons.

File: www/src/ideas/IdeaDetailsForm.tsx [hunk 1]
```diff
@@ -58,11 +85,18 @@ const IdeaDetailsForm: React.FC<Props> = ({idea, setIdea}) => {
...
         <div className={classes.container}>
-            <Header>
-                {t(isNew() ? 'idea.introduceTitle' : 'idea.editTitle')}
-            </Header>
+            <div className={classes.headerContainer}>
+                <Header>
+                    {t(isNew() ? 'idea.introduceTitle' : 'idea.editTitle')}
+                </Header>
+                <IconButton svg={CrossSvg} onClick={navigateBack}/>
+            </div>
```
Comment (aolszewska) [inline www/src/ideas/IdeaDetailsForm.tsx:99]: I'm not sure if the header and the x button should be part of the form component. How do you think about moving them to the idea details component?

File: www/src/ideas/IdeaDetailsForm.tsx [hunk 2]
```diff
@@ -58,11 +85,18 @@ const IdeaDetailsForm: React.FC<Props> = ({idea, setIdea}) => {
...
     return (
         <div className={classes.container}>
-            <Header>
-                {t(isNew() ? 'idea.introduceTitle' : 'idea.editTitle')}
-            </Header>
+            <div className={classes.headerContainer}>
+                <Header>
+                    {t(isNew() ? 'idea.introduceTitle' : 'idea.editTitle')}
+                </Header>
+                <IconButton svg={CrossSvg} onClick={navigateBack}/>
```
Comment (aolszewska) [inline www/src/ideas/IdeaDetailsForm.tsx:98]: Clicking the x button should actually get you to the ideas list. Now, if you just open the link http://localhost:3000/ideas/new and click the x button you'll be taken back to where you have been, not to the list.

File: www/src/ideas/IdeaDetailsForm.tsx (hunk 1 above)
Comment (Milszym) [reply www/src/ideas/IdeaDetailsForm.tsx:99]: Sure, I have moved it to `IdeaDetails` component.

File: www/src/ideas/IdeaDetailsForm.tsx (hunk 2 above)
Comment (Milszym) [reply www/src/ideas/IdeaDetailsForm.tsx:98]: Sure, is `history.push(ROUTE_IDEAS)` fine?

**Comment by Milszym** [comment]
> @Milszym
> 
> > I have added cross icon that navigates back when clicked. I was trying to create svg button, but it is not centered. Have you got any experience in such svg buttons in Material-UI library? I couldn't center the cross icon :/
> > Also I am not sure if wrapping SvgIcon with IconButton is the right way to go with material ui, I couldn't find out another way though :/ What do you think?
> > I was using a pure HTML `img` tag for the svg icons.

Thanks! It worked like a charm.

File: www/src/ideas/IdeaDetailsForm.tsx (hunk 2 above)
Comment (aolszewska) [reply www/src/ideas/IdeaDetailsForm.tsx]: Yes, I think so

**Pull request merged**